### PR TITLE
Dvc 5462 user endpoint

### PR DESCRIPTION
**Pull request by elliotCamblor**
I only added the fields that were actively using to the builder in the endpoint. I can add the rest if we want, it just gets unruly

### CODE REVIEW

File: proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java [hunk 1]
```diff
@@ -28,6 +30,29 @@ public Spec spec() {
         return new Spec();
     }
 
+    @PostMapping("/user")
+    public BaseResponse user(@RequestBody UserRequestBody body, HttpServletResponse response) {
+        try {
+            User sdkUser = User.builder()
+                .userId(body.user_id)
+                .name(body.name)
```
Comment (ajwootto) [inline proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:38]: does this work when these fields are null / aren't set?

File: proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java [hunk 2]
```diff
@@ -28,6 +30,29 @@ public Spec spec() {
...
+                .customData(body.customData)
```
Comment (ajwootto) [inline proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:39]: i think you might as well just list every property here

File: proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java (hunk 1 above)
Comment (elliotCamblor) [reply proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:38]: yep

File: proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java (hunk 2 above)
Comment (elliotCamblor) [reply proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:39]: but its so ugly 😭 okay tho

Comment (ajwootto) [reply proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:39]: thats Java babyyyy

Comment (ajwootto) [reply proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:39]: you could probably write some fancy reflection thing to pull the correct method name based on the name of each property in the request

Comment (elliotCamblor) [reply proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:39]: id rather just get it done lol

File: proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java [hunk 3]
```diff
@@ -28,6 +30,36 @@ public Spec spec() {
...
+                .appVersion(body.appVersion)
+                .appBuild(body.appBuild)
+                .customData(body.customData)
+                .privateCustomData(body.privateCustomData)
+                .createdDate(body.createdDate)
+                .lastSeenDate(body.lastSeenDate)
+                .build();
+
+            var userId = DataStore.Users.size();
+            DataStore.Users.put(userId + "", sdkUser);
```
Comment (nsalamad) [inline proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:50]: Might be more clear to do `var userId = Interger.toString(DataStore.Users.size());`

Comment (elliotCamblor) [reply proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:50]: 👍

**Review by nsalamad — APPROVED** [review]

File: proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java [hunk 4]
```diff
@@ -28,6 +30,36 @@ public Spec spec() {
...
+                .lastSeenDate(body.lastSeenDate)
+                .build();
+
+            var userId = Integer.toString(DataStore.Users.size());
+            DataStore.Users.put(userId, sdkUser);
+
+            var result = new UserData(sdkUser);
+
+            response.addHeader("Location", "user/" + userId);
+            return result;
```
Comment (ajwootto) [inline proxies/java/src/main/java/com/dvc_harness/proxy/controller/ProxyController.java:55]: is the status here 201?

**Review by ajwootto — APPROVED** [review]

**Pull request merged**